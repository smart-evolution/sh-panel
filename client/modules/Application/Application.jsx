// @flow
import React from 'react';
import AlertPanel from 'client/modules/AlertPanel';

type Props = {
  isLoaded: boolean,
  children: React$Element<any>,
  mount: () => void,
};

class Application extends React.PureComponent<Props> {
  componentDidMount() {
    const { mount } = this.props;
    mount();
  }

  render() {
    const { children, isLoaded } = this.props;

    const loader = <div className="c-loader" />;

    return (
      <div className="application">
        {isLoaded && children}
        {!isLoaded && loader}
        <AlertPanel />
      </div>
    );
  }
}

export default Application;
