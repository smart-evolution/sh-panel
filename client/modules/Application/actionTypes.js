export const MOUNT = 'application/mount';
export const LOADED = 'application/loaded';
