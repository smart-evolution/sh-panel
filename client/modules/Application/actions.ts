import * as actionTypes from './actionTypes';

export const mount = () => ({
  type: actionTypes.MOUNT,
});

export const loaded = () => ({
  type: actionTypes.LOADED,
});
