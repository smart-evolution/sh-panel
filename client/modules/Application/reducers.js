// @flow
import * as actionTypes from './actionTypes';

type State = {
  isLoaded: boolean,
};

const defaultState = {
  isLoaded: false,
};

export default function reducers(state: State = defaultState, action: Object) {
  switch (action.type) {
    case actionTypes.LOADED:
      return Object.assign({}, state, {
        isLoaded: true,
      });
    default:
      return state;
  }
}
