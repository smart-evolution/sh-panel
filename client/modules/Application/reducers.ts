import * as actionTypes from './actionTypes';
import type * as types from './types';

type State = {
  isLoaded: boolean;
};

const defaultState = {
  isLoaded: false,
};

export default function reducers(
  state: State = defaultState,
  action: types.Action
) {
  switch (action.type) {
    case actionTypes.LOADED:
      return { ...state, isLoaded: true };
    default:
      return state;
  }
}
