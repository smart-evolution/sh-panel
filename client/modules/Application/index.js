// @flow
import { connect } from 'react-redux';
import Application from './Application';
import * as actions from './actions';
import * as selectors from './selectors';

const mapStateToProps = state => {
  return {
    isLoaded: selectors.getIsLoaded(state),
  };
};

const mapDispatchToProps = dispatch => ({
  mount: () => {
    dispatch(actions.mount());
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Application);
