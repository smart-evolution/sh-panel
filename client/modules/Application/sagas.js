// @flow
import { put } from 'redux-saga/effects';
import * as actions from './actions';

/* eslint-disable import/prefer-default-export */
export function* onApplicationMount(): Iterable<any> {
  yield put(actions.loaded());
}
/* eslint-enable import/prefer-default-export */
