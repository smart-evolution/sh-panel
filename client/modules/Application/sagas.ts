import { put, take, fork, race } from 'redux-saga/effects';
import * as userActions from 'client/models/user/actions';
import * as userActionTypes from 'client/models/user/actionTypes';
import * as agentsSagas from 'client/models/agents/sagas';
import * as agentActions from 'client/models/agents/actions';
import * as agentActionTypes from 'client/models/agents/actionTypes';
import * as agentConfigActions from 'client/models/agentConfigs/actions';
import * as agentConfigActionTypes from 'client/models/agentConfigs/actionTypes';
import * as actions from './actions';

/* eslint-disable import/prefer-default-export */
export function* onApplicationMount(): Iterable<any> {
  yield put(userActions.fetchUser());
  yield take([userActionTypes.LOAD_USER]);
  yield fork(agentsSagas.subscribeOnFetchAgents);
  yield fork(agentsSagas.onFetchAlerts);

  yield put(agentActions.fetchAgents());
  // @ts-ignore - to be fixed
  yield race([
    take([agentActionTypes.LOAD_AGENTS]),
    take([agentActionTypes.FETCH_AGENTS_ERROR]),
  ]);

  yield put(agentConfigActions.fetchAgentConfig());
  // @ts-ignore - to be fixed
  yield race([
    take([agentConfigActionTypes.LOAD_AGENT_CONFIGS]),
    take([agentConfigActionTypes.FETCH_AGENT_CONFIGS_ERROR]),
  ]);

  yield put(actions.loaded());
}
/* eslint-enable import/prefer-default-export */
