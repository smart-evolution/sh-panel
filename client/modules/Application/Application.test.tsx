import _ from 'lodash';
import React from 'react';
import { shallow } from 'enzyme';
import Application from './Application';

describe('Application', () => {
  it('should render with loader', () => {
    const component = shallow(
      <Application isLoaded={false} mount={_.noop}>
        Content
      </Application>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render with children', () => {
    const component = shallow(
      <Application isLoaded mount={_.noop}>
        <p>Some content</p>
      </Application>
    );

    expect(component).toMatchSnapshot();
  });
});
