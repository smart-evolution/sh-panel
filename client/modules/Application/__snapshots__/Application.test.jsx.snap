// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application should render with children 1`] = `
<div
  className="application"
>
  <p>
    Some content
  </p>
  <Connect(AlertPanel) />
</div>
`;

exports[`Application should render with loader 1`] = `
<div
  className="application"
>
  <div
    className="c-loader"
  />
  <Connect(AlertPanel) />
</div>
`;
